lib/config/genconfig/hadeployconfig.go

```diff
@@ -0,0 +1,1102 @@
+	if err != nil {
+		return
+	}
+	c.Config.InitChefServer().InitConfig()
+	c.Config.ChefServer.Config.InstanceCount = fmt.Sprint(chefServerNodeCount)
+
+	hasCustomCerts, hasCustomCertsPerNode, err := c.PromptHaveCustomCerts(CHEF_INFRA_SERVER_NODETYPE)
+	if err != nil {
+		return err
+	}
+
+	c.Config.ChefServer.Config.EnableCustomCerts = hasCustomCerts
+
+	err = c.PromptChefInfraServerPriPubCerts(hasCustomCerts, hasCustomCertsPerNode)
+	if err != nil {
+		return err
+	}
+
+	c.Config.InitExistingInfra().InitConfig()
+	for i := 0; i < chefServerNodeCount; i++ {
+		err = c.PromptChefInfraServerNodeIpCerts(i, hasCustomCerts, hasCustomCertsPerNode)
+		if err != nil {
+			return err
+		}
+	}
+	return
+}
+
+func (c *HaDeployConfigGen) PromptChefInfraServerPriPubCerts(hasCustomCerts, hasCustomCertsPerNode bool) (err error) {
+	if hasCustomCerts && !hasCustomCertsPerNode {
+		priKey, err := c.PromptCert("", CHEF_INFRA_SERVER_NODETYPE, PRI_KEY)
+		if err != nil {
+			return err
+		}
+		c.Config.ChefServer.Config.PrivateKey = priKey
+
+		pubCert, err := c.PromptCert("", CHEF_INFRA_SERVER_NODETYPE, PUB_CERT)
+		if err != nil {
+			return err
+		}
+		c.Config.ChefServer.Config.PublicKey = pubCert
+	}
+	if hasCustomCerts && hasCustomCertsPerNode {
+		c.Config.ChefServer.Config.InitCertsByIP()
+	}
+	return
+}
+
+func (c *HaDeployConfigGen) PromptChefInfraServerNodeIpCerts(i int, hasCustomCerts, hasCustomCertsPerNode bool) (err error) {
+	ip, err := c.PromptNodeIp(fmt.Sprintf(PROMPT_NODE_IP, i+1))
+	if err != nil {
+		return err
+	}
+	c.Config.ExistingInfra.Config.ChefServerPrivateIps = append(c.Config.ExistingInfra.Config.ChefServerPrivateIps, ip)
+	if hasCustomCerts && hasCustomCertsPerNode {
+		priKey, err := c.PromptCert(ip, CHEF_INFRA_SERVER_NODETYPE, PRI_KEY)
+		if err != nil {
+			return err
+		}
+
+		pubCert, err := c.PromptCert(ip, CHEF_INFRA_SERVER_NODETYPE, PUB_CERT)
+		if err != nil {
+			return err
+		}
+
+		nodeCert := config.CertByIP{
+			IP:         ip,
+			PrivateKey: priKey,
+			PublicKey:  pubCert,
+		}
+		*c.Config.ChefServer.Config.CertsByIP = append(*c.Config.ChefServer.Config.CertsByIP, nodeCert)
+	}
+	return
+}
+
+func (c *HaDeployConfigGen) PromptOpenSearchNodes() (err error) {
+	opensearchNodeCount, err := c.Prompt.InputIntRange("No. of OpenSearch Nodes", 3, 100)
+	if err != nil {
+		return
+	}
+	c.Config.InitOpenSearch().InitConfig()
+	c.Config.Opensearch.Config.InstanceCount = fmt.Sprint(opensearchNodeCount)
+
+	hasCustomCerts, hasCustomCertsPerNode, err := c.PromptHaveCustomCerts(OPENSEARCH_NODETYPE)
+	if err != nil {
+		return err
+	}
+
+	c.Config.Opensearch.Config.EnableCustomCerts = hasCustomCerts
+
+	err = c.PromptOpenSearchPubPriCerts(hasCustomCerts, hasCustomCertsPerNode)
+	if err != nil {
+		return err
+	}
+
+	c.Config.InitExistingInfra().InitConfig()
+	for i := 0; i < opensearchNodeCount; i++ {
+		err := c.PromptOpenSearchNodeIpCerts(i, hasCustomCerts, hasCustomCertsPerNode)
```

Comment:
in case of more than 1 node count, the IPs should be unique, we can add checks for that

Reply:
I will consider this as feature enhancement for next time.